### ISSUE TITLE
Implement `roc_fx_fileReadBytes` in CLI platform

### ISSUE THREAD

**Issue by JanCVanB** [roc-lang/roc#4021]
See https://github.com/roc-lang/roc/blob/8fbd09b9ea8d70c896fccb5f56a493bec36e2ce1/examples/interactive/cli-platform/src/lib.rs#L215 for the current placeholder print.

**Comment by JanCVanB** [comment]
I would really like to have file reading by tomorrow - is that possible?

**Comment by JanCVanB** [comment]
Are these errors/warnings related or separate?
```
[jan@framey try-file-io]$ cat test.roc  
app "test"
    packages { pf: "../examples/interactive/cli-platform/main.roc" }
    imports [pf.Stdout, pf.Stderr, pf.Task, pf.File, pf.Path]
    provides [main] to pf

main =
    input <- File.readUtf8 (Path.fromStr "input.txt") |> Task.await
    File.writeUtf8 (Path.fromStr "output.txt") input
[jan@framey try-file-io]$ cat input.txt 
data
[jan@framey try-file-io]$ cat output.txt
cat: output.txt: No such file or directory
[jan@framey try-file-io]$ cargo run test.roc 
    Finished dev [unoptimized + debuginfo] target(s) in 0.18s
     Running `/home/jan/_code/_roc/roc/target/debug/roc test.roc`

── TYPE MISMATCH ─────────────── ../examples/interactive/cli-platform/main.roc ─

Something is off with the type annotation of the main required symbol:

2│      requires {} { main : Task {} [] * }
                             ^^^^^^^^^^^^

This #UserApp.main value is a:

    Task {} [FileReadErr Path.Path File.ReadErr,
    FileReadUtf8Err Path.Path [BadUtf8 Utf8ByteProblem Nat]*,
    FileWriteErr Path.Path File.WriteErr]a [Read [File]*,
    Write [File]*]b ?

But the type annotation on main says it should be:

    Task {} [] [Read [File]*, Write [File]*]b ?




── TYPE MISMATCH ─────────────── ../examples/interactive/cli-platform/main.roc ─

Something is off with the body of the mainForHost definition:

8│  mainForHost : Effect (Result {} []) as Fx
9│  mainForHost = InternalTask.toEffect main
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^

This InternalTask.toEffect call produces:

    Effect.Effect (Result {} [FileReadErr Path.Path File.ReadErr,
    FileReadUtf8Err Path.Path [BadUtf8 Utf8ByteProblem Nat]*,
    FileWriteErr Path.Path File.WriteErr]a) ?

But the type annotation on mainForHost says it should be:

    Fx ?

Tip: Type comparisons between an opaque type are only ever equal if
both types are the same opaque type. Did you mean to create an opaque
type by wrapping it? If I have an opaque type Age := U32 I can create
an instance of this opaque type by doing @Age 23.

────────────────────────────────────────────────────────────────────────────────

2 errors and 2 warnings found in 199 ms.

You can run the program anyway with roc run test.roc
[jan@framey try-file-io]$ 
```

**Comment by JanCVanB** [comment]
FWIW, I've pushed my experimentation branch: https://github.com/roc-lang/roc/tree/try_file_io/try-file-io

**Comment by JanCVanB** [comment]
Fixed by #4030 !
